Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+        TraceError(TAG, L"Unable to find family for font name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return E_INVALIDARG;
+    }
+
+    ComPtr<IDWriteFont> font;
+    HRESULT hr = fontFamily->GetFirstMatchingFont(weight, stretch, style, &font);
+    if (FAILED(hr)) {
+        TraceError(TAG, L"Unable to create font for name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return hr;
+    }
+
+    return font->CreateFontFace(outFontFace);
+}
+
+/**
+ * Gets an informational string as a CFString from a DWrite font face.
+ */
+CFStringRef _DWriteFontCopyInformationalString(ComPtr<IDWriteFontFace> fontFace, DWRITE_INFORMATIONAL_STRING_ID informationalStringId) {
+    if (fontFace == nullptr) {
+        return nullptr;
+    }
+
+    ComPtr<IDWriteLocalizedStrings> name;
+    BOOL exists;
+
+    ComPtr<IDWriteFontFace3> dwriteFontFace3;
+    fontFace.As(&dwriteFontFace3);
+
+    THROW_IF_FAILED(dwriteFontFace3->GetInformationalStrings(informationalStringId, &name, &exists));
+    return exists ? static_cast<CFStringRef>(CFRetain(_CFStringFromLocalizedString(name.Get()))) : nullptr;
+}
+
+/**
+ * Gets a font table as a CFDataRef from a DWrite font face
+ */
+CFDataRef _DWriteFontCopyTable(ComPtr<IDWriteFontFace> fontFace, uint32_t tag) {
+    const void* tableData;
+    uint32_t tableSize;
+    void* tableContext;
+    BOOL exists;
+
+    if (FAILED(fontFace->TryGetFontTable(tag, &tableData, &tableSize, &tableContext, &exists)) || !exists) {
+        TraceError(TAG, L"Unable to get font table for tag %u", tag);
+        return nullptr;
+    }
+
+    // Copy the font table's binary data to a CFDataRef
+    CFDataRef ret = CFDataCreate(kCFAllocatorDefault, reinterpret_cast<const byte*>(tableData), tableSize);
+
+    // As part of the contact for TryGetFontTable, tableContext must always be released via ReleaseFontTable
+    if (tableContext) {
+        fontFace->ReleaseFontTable(tableContext);
+    }
+
+    return ret;
+}
+
+/**
+ * Helper function that gets the slant angle, in degrees, from a DWrite font face.
+ */
+CGFloat _DWriteFontGetSlantDegrees(ComPtr<IDWriteFontFace> fontFace) {
+    ComPtr<IDWriteFontFace1> fontFace1;
+    fontFace.As(&fontFace1);
+
+    DWRITE_CARET_METRICS caretMetrics;
+    fontFace1->GetCaretMetrics(&caretMetrics);
+    if (caretMetrics.slopeRun && caretMetrics.slopeRise) {
+        CGFloat riseOverRun = static_cast<CGFloat>(caretMetrics.slopeRise) / static_cast<CGFloat>(caretMetrics.slopeRun);
+        return (atan(riseOverRun) * 180 / M_PI) - 90.0f; // Degrees returned by this function must be negative
+    } else {
+        // Avoid dividing by 0, skip math for dividing with 0
+        return -0.0f;
+    }
+}
+
+/**
+ * Gets the accumulated bounding box of all the glyphs in a DWrite font face, as a CGRect
+ * Returns in terms of design units
+ */
+CGRect _DWriteFontGetBoundingBox(ComPtr<IDWriteFontFace> fontFace) {
```

Comment:
>ComPtr<IDWriteFontFace> fontFace [](start = 33, length = 32)

:-( This is a bad pattern to use. In C++ everything is pass by value by default. This means that you are passing a ComPtr<> by value which means that you making a copy of a ComPtr object and unnecessarily doing an extra AddRef Release on it. 

Please use const ComPtr<>& as I hoped you had gotten used to in previous projects with shared_ptr ...